server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -711,7 +714,8 @@ private IndexMetadata(
         final int routingPartitionSize,
         final ActiveShardCount waitForActiveShards,
         final Map<String, RolloverInfo> rolloverInfos,
-        final boolean isSystem
+        final boolean isSystem,
+        final int indexTotalShardsPerNodeLimit
```

Comment:
Will this not break the consumers of this ctor?

Reply:
The only consumer of this constructor is the builder which is updated accordingly

Reply:
Wondering why does the ctor need to have this, shouldn't this be optional anyways?